### DISCUSSION TITLE
[Idea] To increase the service catalog with glpi agent Hillstone Stone OS

### DISCUSSION THREAD

**Discussion by sistemmsn** [glpi-project/glpi-agent#710] [Ideas]
### Your idea

We are a Hillstone partner and I would like to have the inventory data of our brand
https://github.com/glpi-project/sysobject.ids/issues/34

**Comment by g-bougard** [comment]
Hi @sistemmsn 

we are always happy to add more devices support.

But have you still tested glpi-agent with that devices ?

Have you tried yourself to add the required line in your sysobject.ids file ? You have a documentation about what can be set in that file here: https://glpi-agent.readthedocs.io/en/latest/database.html#snmp-device-ids-database
The most important thing is the exact expected manufacturer string.
If only networking devices have to be inventoried, the type field just need to be `NETWORKING`.
If the model can't be discovered in standard OID, we may need to know if the sysObjectID is changing by deveice model. If yes, an entry by model can be added. If the devices have useful private OID, a dedicated module can be developed in that folder: https://github.com/glpi-project/glpi-agent/tree/develop/lib/GLPI/Agent/SNMP/MibSupport

MIB files can be useful and also [snmpwalk output in the expected format](https://glpi-agent.readthedocs.io/en/latest/bug-reporting.html#snmpwalk-output) can help to validate the expected result.

**Reply by sistemmsn** [reply]
I've already tried it and it doesn't collect any data like you're like a cisco oh a fotinet,

Do what the command mentions except that I had to cut it because it grew a lot. I don't know if it's normal.
run: snmpwalk -v <version> -c <community> -t 15 -Cc -On -Ox <somehost> .1 > walk-filename.walk
[walk-filename.zip](https://github.com/user-attachments/files/16183423/walk-filename.zip)

and this is what the glpi agent gives me
```
PS C:\Program Files\GLPI-Agent> .\glpi-netdiscovery.bat --host 10.0.0.1 --credentials version:2c,community:example --debug -i -s "C:\Program Files\GLPI-Agent"
netdiscovery XMLs will be saved in: C:\Program Files\GLPI-Agent/netdiscovery
netinventory XMLs will be saved in: C:\Program Files\GLPI-Agent/netinventory
====
[debug] initializing job 1
[debug] initializing block 10.0.0.1-10.0.0.1
[debug] Current netdiscovery run expiration timeout: 15 minutes
[debug] using 1 netdiscovery worker
[debug] starting job 1 with 1 ip to scan using 1 worker
[debug] #1, scanning 10.0.0.1
[debug] #1, - scanning 10.0.0.1 with SNMP, credentials 1: no result, no response from host 10.0.0.1
[debug] #1, - scanning 10.0.0.1 with netbios: no result
[debug] #1, - scanning 10.0.0.1 with echo ping: success
[debug] #1, - scanning 10.0.0.1 in arp table: success
[info] #1, Netdiscovery result for 10.0.0.1 saved in C:/Program Files/GLPI-Agent/netdiscovery/10.0.0.1.xml
[debug] #1, #1, worker termination
[debug] #1, Netdiscovery worker terminated
```

**Reply by g-bougard** [reply]
Hi @sistemmsn 

it seems your walk is looping on some OIDs... this is a device problem. You should contact the device manufacturer, maybe a new firmware exists to fix this problem. The problem here with your walk is we can't know if this prevents some other important OID to be exposed and used by the agent.

I don't know also if this can be a problem for a live analysis by the agent.

Regarding you command run, "no result, no response from host" means you probably didn't use the right credentials.

**Reply by sistemmsn** [reply]
I don't know, Rick, I doubt that's what it is. Look.

![imagen](https://github.com/user-attachments/assets/b8231a74-8624-4346-9cb9-4e3321e61d07)

![imagen](https://github.com/user-attachments/assets/993720fd-778c-4543-9f23-ef55a93ddeff)

**Reply by g-bougard** [reply]
Hi @sistemmsn 

first check your agent can reach the 161 udp port of the device. Maybe a firewall is just blocking all the requests: I see you "nagios" server has access to the device, but this doesn't mean the computer where you installed the agent has.

If this is not a firewall problem, the device may be too long to answer: in that case, try to use `--timeout` option to specify a longer timeout, the default is 1 second.

If there's a network problem or the device doesn't always send an answer, you also can try the `--retry` option (new option since glpi-agent v1.9). Are you using the latest version v1.10 ?

**Reply by sistemmsn** [reply]
You're right it was layer 8 hahaha but you don't know the manufacturing

![imagen](https://github.com/user-attachments/assets/7fca9fc4-7b0c-4983-9e46-109e4071cf31)

**Reply by g-bougard** [reply]
So this time, you can add the following lines to your sysobject.ids file:
```
28557	Hillstone	NETWORKING
28557.1.334	Hillstone	NETWORKING	SG-6000-A1100
```
Be careful, the fields separator in that file is one tabulation, don't use spaces.
You can put these lines anywhere in the file.
One big question remains if `private.28557.1.334` OID really identify one Hillstone model or a model brand. If it's a model `SG-6000-A1100` is suitable if this is a brand `Hillstone Security Appliance` is more convenient and then the model have to be extracted somewhere else, maybe using a dedicated MibSupport module, and then I need a full snmpwalk without looping and/or a MIB file if you can have it.

**Reply by sistemmsn** [reply]
everything is here also make the request with centreon
https://github.com/centreon/centreon-plugins/issues/4937
If you need more information, I can gladly request it from Marca.

**Reply by g-bougard** [reply]
Hi @sistemmsn 

I also took some time to make more research.

I was able to produce a MibSupport/Hillstone.pm module: [Hillstone.pm.zip](https://github.com/user-attachments/files/16263515/Hillstone.pm.zip)

Can you add it to your glpi-agent installation and test it ?

In theory, you finally don't need to update sysobjectid.ids file: the only benefit would be to force the model if it is not well discovered. Actually, I decided to look for the model in sysDescr OID taking the last string of the sysDescr. It seems correct for the only one model walk you provided, but I'm definitively not sure it's the right way to discover it. That's a share there's no model OID, maybe you can request for it in future OS updates as a customer or Hillstone partner ;-)

I saw the OID for serialnumer & firmware was changed in MIBs v3. So this module will fallback on the older OID if the more recent gives nothing: don't know if its relevant, but older Hillstone devices should also be supported.

Can you confirm this module makes the job before I add it to the develop branch ?

**Comment by sistemmsn** [comment]
Hey, I have a question about Hillstone, like any brand, there are many models of firewalls, how can I send them to you? Oh, did I ask the manufacturer?

**Reply by g-bougard** [reply]
Hi @sistemmsn 

manufacturer shouldn't change OIDs for main interesting fields. So we can expect the module will support almost every models, even in the future. If something changes and makes the module no more compatible, we will have to update the module sending some shame on Hillstone ;-)

If you still find differences, the module should be updated before released.

Have you tested the module I proposed you last week in my previous comment ?
